src/lib/chips/chip.ts

```diff
@@ -1,17 +1,154 @@
-import { Component, ElementRef, Renderer } from '@angular/core';
+import {
+  // Classes
+  AfterViewInit,
+  Component,
+  ElementRef,
+  EventEmitter,
+  OnDestroy,
+  Renderer,
 
-@Component({
-  selector: 'md-chip, [md-chip]',
+  // Functions
+  forwardRef
+} from '@angular/core';
+
+import {MdFocusable} from '../core/a11y/list-key-manager';
+
+/*
+ * A reusable component config. Utilized by the MdBasicChip
+ * as well as the child class MdChip.
+ */
+export const MD_BASIC_CHIP_COMPONENT_CONFIG: Component = {
+  selector: 'md-basic-chip, [md-basic-chip]',
   template: `<ng-content></ng-content>`,
   host: {
     // Properties
-    'class': 'md-chip',
     'tabindex': '-1',
-    'role': 'option'
-  }
-})
-export class MdChip {
+    'role': 'option',
+
+    // Attributes
+    '[attr.disabled]': 'disabled',
+    '[attr.aria-disabled]': 'isAriaDisabled',
+
+    // Events
+    '(click)': 'click($event)',
+  },
+  inputs: ['disabled'],
+  outputs: ['didfocus', 'destroy']
+};
+
+/**
+ * The base, unstyled chip component extended by the styled MdChip and any custom chips.
+ */
+@Component(MD_BASIC_CHIP_COMPONENT_CONFIG)
+export class MdBasicChip implements MdFocusable, OnDestroy {
+
+  /* Whether or not the chip is disabled. */
+  protected _disabled: boolean = false;
+
+  /**
+   * Emitted when the chip receives focus.
+   * @type {EventEmitter}
+   */
+  public didfocus = new EventEmitter();
+
+  /**
+   * Emitted when the chip is destroyed.
+   * @type {EventEmitter}
+   */
+  public destroy = new EventEmitter();
+
   constructor(protected _renderer: Renderer, protected _elementRef: ElementRef) {}
 
-  ngAfterContentInit(): void {}
+  ngOnDestroy(): void {
+    this.destroy.emit({chip: this});
+  }
+
+  /**
+   * Whether or not the chip is disabled.
+   *
+   * @returns {boolean}
+   */
+  get disabled(): boolean {
+    return this._disabled;
+  }
+
+  /**
+   * Sets the disabled state of the chip.
+   *
+   * @param value `true` to disable, `false` to enable.
+   */
+  set disabled(value: boolean) {
+    this._disabled = (value === false || value === undefined) ? null : true;
+  }
+
+  /**
+   * A String representation of the current disabled state.
+   *
+   * @returns {String}
+   */
+  get isAriaDisabled(): string {
```

Comment:
This should be internal (start with an underscore but not use the `private` keyword)